packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -225,8 +254,22 @@ export const addParticipant = async (req: Request, res: Response) => {
       giveaway.participants = giveaway.participants.filter(
         (p) => p.id !== participant.id
       );
+      const { id, title, description, prize, rules, image } = giveaway;
       await giveaway.save();
       throw error;
+      await GiveawayLog.create({
+        action: 'add-giveaway-participant',
+        author: req.user.email,
+        giveaway_id: id,
+        changes: {
+          title,
+          description,
+          prize,
+          rules,
+          image,
+        },
+      });
```

Comment:
I think this should not be here. If we reach this catch, it means that the user was approved but for some reason it failed to add it to the contract, so the user was not added. Also, we probably should discuss if there is any advantage of logging the participant addition, since the user itself it's the only one that can take this action (the admins cannot add participants), so this will just pollute the logs I would say

Reply:
Good catch, and since only the user can add participants, there's probably no valuable information to be withdrawn from this change 👍